src/controls/MenuSelectTableBorderStyle.tsx

```diff
@@ -0,0 +1,193 @@
+import BorderStyle from "@mui/icons-material/BorderStyle";
+import { ClickAwayListener, Fade, Paper, Popper } from "@mui/material";
+import * as React from "react";
+import { makeStyles } from "tss-react/mui";
+import { useRichTextEditorContext } from "../context";
+import {
+  BorderStyleDashed,
+  BorderStyleDotted,
+  BorderStyleDouble,
+  BorderStyleGroove,
+  BorderStyleInset,
+  BorderStyleNone,
+  BorderStyleOutset,
+  BorderStyleRidge,
+  BorderStyleSolid,
+} from "../icons";
+import { Z_INDEXES } from "../styles";
+import MenuButton from "./MenuButton";
+import type { MenuButtonTooltipProps } from "./MenuButtonTooltip";
+import MenuControlsContainer from "./MenuControlsContainer";
+
+export type BorderStyleSelectOption = {
+  value: string;
+  IconComponent: React.ElementType<{
+    className: string;
+  }>;
+  label?: string;
+  shortcutKeys?: MenuButtonTooltipProps["shortcutKeys"];
+};
+
+export type MenuSelectTableBorderStyleProps = {
+  options?: BorderStyleSelectOption[];
+  label?: string;
+  className?: string;
+};
+
+const useStyles = makeStyles({ name: { MenuSelectTableBorderStyle } })(
+  (theme) => ({
+    root: {
+      zIndex: Z_INDEXES.BUBBLE_MENU,
+    },
+
+    paper: {
+      backgroundColor: theme.palette.background.default,
+    },
```

Comment:
This leads to a lint warning currently, since it's not used: https://github.com/sjdemartini/mui-tiptap/actions/runs/7368820861/job/20510696875?pr=185#step:10:12

(I will probably update eslint so that it returns a failing status code even for warnings, since ideally CI would flag this more noisily and prevent warnings from being merged.)